utilities/dataflow_pubsub_to_hcapi/dataflow_pubsub_to_hcapi.py

```diff
@@ -0,0 +1,461 @@
+            rawMessage = dict()
+            rawMessage['rawMessage'] = message
+            rawMessage['errorMessage'] = errorMessage
+            processFlag = "fail"
+            return rawMessage,processFlag
+        else:
+            return patientResourceJSON,processFlag
+            
+
+#Below class is used to consume messages from Pub/Sub and transform it
+class consumeTransformMessages(beam.DoFn):
+    """ Beam DoFn to transform raw data and create FHIR resource as per Transformation"""
+    #beam output tag to mark transformed element as pass record or fail record
+    failTag = "failRecord"
+    successTag = "successRecord"
+    
+    def process(self,element):
+        """
+        Note:- You can reuse this class to add/remove attributes based on payload receieved via pub/sub
+        process to consume messages from pub/sub subscription and construct raw message JSON with attributes.
+        There are multiple logging statements in place for debugging purpose one can remove as per requirements
+        :return: transformed raw JSON
+        :rtype: dict
+        """
+        try:
+            logging.info("printing raw element from pubsub")
+            logging.info(f"{element}")
+            logging.info("printing raw data element from pubsub")
+            rawMessage = json.loads(element.data.decode('utf-8'))
+            logging.info(json.dumps(rawMessage))
+            logging.info("Printing attributes from pubsub message")
+            messageAttributes = element.attributes
+            logging.info(json.dumps(messageAttributes))
+            rawMessage['topic'] = messageAttributes['topic']
+            rawMessage['timestamp'] = messageAttributes['timestamp']
+            yield pvalue.TaggedOutput(self.successTag,rawMessage)
+        except Exception as error :
+            logging.error("process to transform pub/sub record record failed due to : {}".format(error, exc_info=True))
+            errorMessage = "process to transform pub/sub record record failed due to : {}".format(str(type(error).__name__)+" --> "+ str(error))
+            transformedPayload = dict()
+            transformedPayload['errorMessage'] = errorMessage
+            transformedPayload['rawMessage']= str(element)
+            yield pvalue.TaggedOutput(self.failTag,transformedPayload)
+    
+class writeToGCS(beam.DoFn):
+    """ Class to write raw and errored messages to GCS """
+    def process(self, element, bucket, messageType, topic=None):
+        """
+         write raw / error messages to respective bucket
+        :param str element: PCollection of message from convert kafka record function
+        :param str bucket: bucket name
+        :param str messageType: message type raw or error message
+        :return: None message written to cloud storage
+        :rtype: None
+        """
+        #TO DO below logging only for testing purpose delete when deploying to prod
```

Comment:
remove this

Reply:
Though the function returns None its good to keep so that users can understand input and output for the function.

Reply:
I was referring to the TODO here on line 329